In [1]:
################################### DOXX2TXT
################################### DOCX

In [109]:
import get_file_word
fileDir = r"/Users/wpessoa/repositorios/Fluxo_Editorial/words/"
fileExt = r".docx"
get_file_word_info_info(fileDir,fileExt)


NameError: name 'get_file_word_info_info' is not defined

In [45]:
# Importar bibliotecas
from os import listdir
from os.path import isfile, join
import docx2txt
import docx
import os
import glob

from os import listdir
from os.path import isfile, join

In [94]:
import glob
targetPattern = r"/Users/wpessoa/repositorios/Fluxo_Editorial/words/*.docx"
files_array = glob.glob(targetPattern)

In [107]:
fileDir = r"/Users/wpessoa/repositorios/Fluxo_Editorial/words/"
fileExt = r".docx"
[_ for _ in os.listdir(fileDir) if _.endswith(fileExt)]

['ID233867-06_12_2019_EFEITOS DA GESTÃO SOBRE A PRODUTIVIDADE - PARA PUBLICAÇÃO_Final_19122019_155125.docx',
 'ID233885-1- Metodol Rec Did - rev.docx',
 'ID233886-2- Metodol LD - rev.docx',
 '~$233867-06_12_2019_EFEITOS DA GESTÃO SOBRE A PRODUTIVIDADE - PARA PUBLICAÇÃO_Final_19122019_155125.docx']

In [105]:
# definir uma variável para o local onde estão os arquivos Word
mypath="/Users/wpessoa/repositorios/Fluxo_Editorial/words/"
#mypath='/Users/walnerpessoa/repositories/Fluxo_Editorial/words'

# gerar vetor com ons nomes dos arquivos Word
onlyfiles = [f for f in listdir(mypath) if isfile(join(mypath, f))]

# verificar se existe o diretório
os.path.exists('/Users/wpessoa/repositorios/Fluxo_Editorial/words') 

True

In [106]:
print(onlyfiles)

['ID233867-06_12_2019_EFEITOS DA GESTÃO SOBRE A PRODUTIVIDADE - PARA PUBLICAÇÃO_Final_19122019_155125.docx', 'ID233885-1- Metodol Rec Did - rev.docx', 'ID233886-2- Metodol LD - rev.docx', '~$233867-06_12_2019_EFEITOS DA GESTÃO SOBRE A PRODUTIVIDADE - PARA PUBLICAÇÃO_Final_19122019_155125.docx']


In [97]:
indice = 0
image_array=[]
table_chunks = []

# abrir conecção com o arquivo Word 
doc = docx.Document(files_array[indice])

# data criação
dt_doc = doc.core_properties.created
# ler em cada parágrafo dentro do arquivo Word
paragra= [p.text for p in doc.paragraphs]

# extai a quantidade de caracteres
caracteres = docx2txt.process(files_array[indice])

# gerar um array com tamanho de todas as imagens
for image in doc.inline_shapes:
    image_array.append([image.width, image.height])
    # print (image.width, image.height)
    
# gerar um array com todas as tabelas
for table in doc.tables:
    table_chunks.append(table)

#Saída dos dados
print('Nome do diretório: ',mypath)
print('Nome do arquivo: ',files_array[indice])
print("Data de Criação do Arquivo: ", dt_doc,"\n")
print('Quantidade de carcteres:',len(caracteres))
print("Quantidade de parágrafos: ", len(paragra))
print("Quantidade de imagens no vetor de imagens/gráficos: ",len(image_array))
print("Quantidade de Tabelas: ",len(table_chunks))


Nome do diretório:  /Users/wpessoa/repositorios/Fluxo_Editorial/words/
Nome do arquivo:  /Users/wpessoa/repositorios/Fluxo_Editorial/words/ID233867-06_12_2019_EFEITOS DA GESTÃO SOBRE A PRODUTIVIDADE - PARA PUBLICAÇÃO_Final_19122019_155125.docx
Data de Criação do Arquivo:  2019-12-06 13:09:00 

Quantidade de carcteres: 59270
Quantidade de parágrafos:  356
Quantidade de imagens no vetor de imagens/gráficos:  11
Quantidade de Tabelas:  3


Nome do diretório:  /Users/wpessoa/repositorios/Fluxo_Editorial/words/
Nome do arquivo:  ID233867-06_12_2019_EFEITOS DA GESTÃO SOBRE A PRODUTIVIDADE - PARA PUBLICAÇÃO_Final_19122019_155125.docx
Quantidade de carcteres: 59270

Quantidade de parágrafos:  356
Quantidade de imagens no vetor de imagens:  11
Quantidade de Tabelas:  3
Data de Criação do Arquivo:  2019-12-06 13:09:00

In [98]:
#####

In [99]:
#teste com metodos da biblioteca Docx

print(doc.paragraphs[0].runs)
print(doc.paragraphs[0].text)
print(doc.paragraphs[0].runs[0].text)
print(doc.paragraphs[0].runs[0].style)
#print(doc.paragraphs[1].runs[0].text)
#print(doc.paragraphs[1].runs[0].style)
print(doc.core_properties.created)
print(doc.inline_shapes)
print(doc.inline_shapes[0].type)
print(doc.inline_shapes[0].height)
print(doc.inline_shapes[0].width)

#


[<docx.text.run.Run object at 0x7f9ecef827c0>, <docx.text.run.Run object at 0x7f9ecef828b0>, <docx.text.run.Run object at 0x7f9ed1f8c100>, <docx.text.run.Run object at 0x7f9ed1f8c580>]
EFEITOS DA GESTÃO SOBRE A PRODUTIVIDADE DAS EMPRESAS 
EFEITOS DA GESTÃO
2019-12-06 13:09:00
12
2898000
5376328
